services/QuillCMS/lib/tasks/reindex_responses.rake

```diff
@@ -0,0 +1,8 @@
+namespace :elasticsearch do
+  desc 'Reindex Responses in Elasticsearch'
+  task :build_response_index, [:batch_size] => [:environment] do |task, args|
+    (1..Response.last.id).step(args[:batch_size].to_i).each do |starting_index|
+      BulkReindexResponsesWorker.perform_async(starting_index, args[:batch_size].to_i - 1)
```

Comment:
Let's pass the full batch size into the job. I think it will be surprising that this rake task does `args[:batch_size].to_i -1` before passing it along, e.g. the rake task is sent `batch_size` = 100, and then pass the job a `batch_size` = 99.

Reply:
Agreed that there should be a run this rake task without running it for all responses for testing. Might be worth having `starting_index` and ending_index` params in case you need to stop it and start it again.

Reply:
> You could use [find_each](https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/Batches.html#method-i-find_each) and pass the batch size as a parameter in there. This would optimize the querying by ensuring getting the full batch size each iteration (`starting_index..(starting_index+batch_size)` just ensures that you're getting at most batch size records)
> 
> I would probably use the rake task to call a `MainBulkReindexResponsesWorker` that spawns all of the smaller `BulkReindexResponsesWorker` that way the whole thing is running in Sidekiq. Also, for testing purposes, you probably need some kind of parameter for `MainBulkReindexResponsesWorker` that sets a limit on the percent of responses that you reindex.

For simplicity's sake I changed it to just the one worker that loops through the responses and sends bulk requests for each batch. Testing this on staging now to see how this goes.